common/src/main/java/com/wynntils/core/text/StyledText.java

```diff
@@ -355,6 +377,51 @@ int addHoverEvent(HoverEvent hoverEvent) {
         return hoverEvents.size();
     }
 
+    public int length() {
+        return parts.stream().mapToInt(StyledTextPart::length).sum();
+    }
+
+    public Optional<JsonObject> toJsonObject() {
+        String json = StringUtils.substringBeforeLast(getString(), "}") + "}"; // remove extra unnecessary info
+        try {
+            return Optional.of(JsonParser.parseString(json).getAsJsonObject());
+        } catch (JsonSyntaxException e) {
+            return Optional.empty();
+        }
+    }
+
+    public PartStyle getStyleAt(int index) {
+        for (StyledTextPart part : parts) if ((index -= part.length()) < 0) return part.getPartStyle();
+        return PartStyle.NONE;
+    }
```

Comment:
This is logically wrong I think, but I also don't see any case where this should be used. I checked the usages and I think I can find a better solution for this later